src/gui/creds/shibboleth/shibbolethwebview.cpp

```diff
@@ -74,10 +75,24 @@ ShibbolethWebView::ShibbolethWebView(AccountPtr account, QWidget* parent)
     connect(page->networkAccessManager()->cookieJar(),
             SIGNAL(newCookiesForUrl (QList<QNetworkCookie>, QUrl)),
             this, SLOT(onNewCookiesForUrl (QList<QNetworkCookie>, QUrl)));
+
     page->mainFrame()->load(account->url());
     this->setPage(page);
     setWindowTitle(tr("%1 - Authenticate").arg(Theme::instance()->appNameGUI()));
 
+    // Debug view to display the cipher suite
+    if( !qgetenv("OWNCLOUD_SHIBBOLETH_DEBUG").isEmpty() ) {
+        // open an additional window to display some cipher debug info
+        QWebPage *debugPage = new UserAgentWebPage(this);
+        account->lendCookieJarTo(debugPage->networkAccessManager());
+        debugPage->mainFrame()->load( QUrl("https://cc.dcsec.uni-hannover.de/"));
```

Comment:
Why do we need an external web service for that? we can query for the supported cipher suites, see http://doc.qt.io/qt-4.8/qsslconfiguration.html#ciphers.